tfsdk/config.go

```diff
@@ -21,10 +21,45 @@ func (c Config) Get(ctx context.Context, target interface{}) diag.Diagnostics {
 	return reflect.Into(ctx, c.Schema.AttributeType(), c.Raw, target, reflect.Options{})
 }
 
-// GetAttribute retrieves the attribute found at `path` and returns it as an
+// GetAttribute retrieves the attribute found at `path` and populates the
+// `target` with the value.
+func (c Config) GetAttribute(ctx context.Context, path *tftypes.AttributePath, target interface{}) diag.Diagnostics {
+	attrValue, diags := c.getAttributeValue(ctx, path)
+
+	if attrValue == nil || diags.HasError() {
```

Comment:
I'm still 🤔 about the case where `attrValue` is `nil`. If error diagnostics are returned, the provider will think that `target` has been set, when it has not. If error diagnostics _are_ returned, then the check is unnecessary. The only case in which the check helps us avoids a panic but is really hard to debug.

Do we want to catch a `nil` `attrValue` and return an error diagnostic for it, instead?